src/Symfony/Component/Translation/Bridge/Lokalise/Tests/LokaliseProviderFactoryTest.php

```diff
@@ -32,6 +34,23 @@ public function incompleteDsnProvider(): iterable
         yield ['lokalise://default'];
     }
 
+    public function testBaseUri()
+    {
+        $httpClient = $this->createMock(MockHttpClient::class);
```

Comment:
```suggestion
        $httpClient = $this->createMock(HttpClientInterface::class);
```

Reply:
Oh no.. here I am trying to help but I only make things worse. 

I committed this, it should be reverted. 
-----

What one can do is to instantiate a new MockClient. Then make the LokaliseProvider to send a request. Then we run assertion on the request. 

```php
// Psudo code. 

$client = new MockHttpClient([new Response(200)]);
$factory = new LokaliseProviderFactory($httpClient, $this->getLogger(), $this->getDefaultLocale(), $this->getLoader());

// Make Lokalise send a HTTP request
$request = $client->getLastRequest(); // Or a similar named method
$this->assertSame('https://api.lokalise.com/api2/projects/PROJECT_ID/', $request->getUri());
```

Reply:
HttpClientInterface doesn't know about `withOptions` method, so we can't use it to create the mock.

@Nyholm I can't find any similar method like `getLastRequest`, are you sure it exists in HttpClient?

Reply:
> HttpClientInterface doesn't know about withOptions method, so we can't use it to create the mock.

Yes. That is why the change I committed (suggested by Oskar) was wrong. 

---

> @Nyholm I can't find any similar method like getLastRequest, are you sure it exists in HttpClient?

Oops, Im slightly wrong again. I've checked the sources now.. =)
You create a `MockResponse` and after the request is "sent" the `Response` object will have URL, HTTP method and response options. 

```php
// Psudo code. 

$response = new MockResponse('body');
$client = new MockHttpClient([$response]);
$factory = new LokaliseProviderFactory($httpClient, $this->getLogger(), $this->getDefaultLocale(), $this->getLoader());

// Make Lokalise send a HTTP request

$this->assertSame('https://api.lokalise.com/api2/projects/PROJECT_ID/', $response->getRequestUrl());
```

Reply:
My last commit should work better. Thanks for the tip to do a real HTTP request